### ISSUE TITLE
建议在计算loss的时候，增加两个tensor维度是否一致进行检测。

### ISSUE THREAD

**Issue by wangyexiang** [PaddlePaddle/Paddle#48544]
### 需求描述 Feature Description

在阅读官方文档的时候，在房价预测的例子中，敲代码的时候：
```
x = training_data[:, :-1]
y = training_data[:, -1:]#这里
xx = paddle.to_tensor(x)
yy = paddle.to_tensor(y)
predicts = model(xx)
Loss = loss(predicts, yy)
```
不小心写成了：
```
x = training_data[:, :-1]
y = training_data[:, -1]#这里
xx = paddle.to_tensor(x)
yy = paddle.to_tensor(y)
predicts = model(xx)
Loss = loss(predicts, yy)
```
结果在计算的过程没任何警告，导致计算出错。

### 替代实现 Alternatives

因此，建议在计算loss的时候，给出一个维度不一致的警告信息，方便bug的查找。

**Comment by cuicheng01** [comment]
您好，请问这里的loss是什么loss呢？或者麻烦给出一下这个demo的链接～

**Comment by wangyexiang** [comment]
> 您好，请问这里的loss是什么loss呢？或者麻烦给出一下这个demo的链接～

您好，这个loss是`loss = MSELoss()`，原demo的地址：[https://www.paddlepaddle.org.cn/tutorials/projectdetail/4309126](url)。

**Comment by cuicheng01** [comment]
好的，这里本身会做一个广播的操作，所以不会报错，不过你提的这个问题我可以反馈下～

**Comment by wangyexiang** [comment]
> 好的，这里本身会做一个广播的操作，所以不会报错，不过你提的这个问题我可以反馈下～

嗯，多谢。给出一个警告信息还是有必要的。
